src/components/studies/StudyBuilder.tsx

```diff
@@ -293,9 +284,19 @@ const StudyBuilder: FunctionComponent<StudyBuilderProps> = ({
                         hasObjectChanged={hasObjectChanged}
                         saveLoader={saveLoader}
                         id={id}
-                        onUpdate={(_section: StudySection, data: any) => {
-                          console.log(_section)
-                          // moveToNextSection(_section)
+                        currentAppDesign={
+                          builderInfo.study.clientData.appDesign ||
+                          ({} as StudyAppDesign)
+                        }
+                        onSave={() => {
+                          saveStudy(builderInfo.study)
+                        }}
+                        onUpdate={(data: StudyAppDesign) => {
+                          setHasObjectChanged(true)
+                          builderInfo.study.clientData.appDesign = data
```

Comment:
you don't want to do it like this. because you re modifying your builderInfo twice: once lne 296 and then in setData. Something like this would work better:
```  
onUpdate={(data: StudyAppDesign) => {
                          setHasObjectChanged(true)
                          const updatedStudy = {...builderInfo.study}
                          updatedStudy.clientData.appDesign = data
              
                          setData({
                            ...builderInfo, study:  updatedStudy
                          })
         ```